runtime/common/src/lib.rs

```diff
@@ -139,12 +139,19 @@ where
 {
 	let mut pool = pallet_pool_system::Pool::<T>::get(pool_id)
 		.ok_or(pallet_pool_system::Error::<T>::NoSuchPool)?;
-	let (nav_loans, _) =
-		pallet_loans::Pallet::<T>::update_portfolio_valuation_for_pool(pool_id, price_input)?;
+
+	let nav_loans =
+		pallet_loans::Pallet::<T>::update_portfolio_valuation_for_pool(pool_id, price_input)
+			.map(|(nav_loans, _)| nav_loans)
+			.unwrap_or_else(|| pallet_loans::Pallet::<T>::portfolio_valuation(pool_id).value());
```

Comment:
Note that `update_portfolio_valuation_for_pool()` overrides the `PortfolioValuation` storage, so calling `poltfolio_valuation()` will not return any different from what the first call has set.

You need to call `portfolio_valuation()` before line 144.

Reply:
The unwrap only takes place if the update failed, right?

Reply:
`update_portfolio_valuation_for_pool` is not transactional, and it can fail, leaving some storage written. Right now, seeing the implementation, it's not the case, but we can break this required "invariant" silently.

Reply:
> so calling poltfolio_valuation() will not return any different from what the first call has set.

But yeah, what I said regarding this is not true. Although could be true in a future refactor

Reply:
So what would you suggest?

Reply:
Something like:
```rust
let prev_nav_loans = pallet_loans::Pallet::<T>::portfolio_valuation(pool_id).value();
let nav_loans = pallet_loans::Pallet::<T>::update_portfolio_valuation_for_pool(pool_id, price_input)
    .map(|(nav_loans, _)| nav_loans)
    .unwrap_or_else(|_| prev_nav_loans);
```

Reply:
fixed @lemunozm